### DISCUSSION TITLE
How do you perform horizontal scrolling without shift key ?

### DISCUSSION THREAD

**Discussion by sorasful** [tailwindlabs/tailwindcss#7869] [Help]
Hello there,

I have horizontal scrolling working nicely, but when I want to scroll I need to press `shift key`, which is not very natural.
I did not find any resources saying how to be able to scroll without having to press the `shift key`.

Thanks for your time :pray: 

```
                    <div className="px-4 flex gap-4 max-w-xl  lg:max-w-4xl overflow-x-scroll no-scrollbar py-5 xlg:ml-28">
                           {
                                  // my items here
                           }
                    </div>
```

**Comment by christhofer** [comment]
There are some mice that supports horizontal scrolling, some have an additional scroll wheel in the left side, or Apple Magic Mouse that use swipe gesture for scrolling. Obviously not everyone have these kind of mouse.

For better UX, you can use left & right button + a bit of javascript to scroll the content. Example: amazon.com & netflix.com
![image](https://user-images.githubusercontent.com/13524958/158768902-3866e9e9-5be5-4ba8-89c9-2e868130445b.png)
![image](https://user-images.githubusercontent.com/13524958/158769667-4f2fc2c0-01d9-4066-a4b4-d83e8bc65226.png)

Or you can use javascript to add mouse drag functionality, there's plenty library to add this functionality, one that I'm currently using in my projects is swiperjs. Can't find any example that use this solution as mouse drag is also not convenient.

**Reply by sorasful** [reply]
Adding the left and right arrow might work, this could be an alternative, but I'm really trying to check if there's something I can do just with my scrolling wheel. Thanks for your answer !

**Comment by tlgreg** [comment]
There is no real css-only solution to change the default scroll direction. There is a "hack" using CSS transforms, but it has its limitations and I wouldn't use it outside of experiments ([CSS-Tricks article](https://css-tricks.com/pure-css-horizontal-scrolling/)).

It's better to have a javascript solution that doesn't block native horizontal scrolling (shift+scroll, etc).
Be that buttons on the sides or desktop simulated horizontal dragging or simply changed default scroll direction with javascript.
Which one is better, depends on the intended design and behavior of the site/app.
Using buttons has probably the best UX for the most common situations like in @christhofer's examples, where vertical scrolling is still required on the page.

**Comment by ederbiason** [comment]
Did you find a solution? I looking for a way to make this work, but no success yet

**Reply by sorasful** [reply]
Unfortunately, no @ederbiason  :/    
I added a little scrollbar at the bottom, so people can use this scrollbar to move around.

**Reply by ederbiason** [reply]
Oh, I searched a lot but can't find nothing... I will do something like you

**Reply by alexredder** [reply]
have you found a solution? - i want users to simply scroll down on their mouse but as they come to the list-section scroll through horizontally and then after continue vertically.. i think i saw it somewhere but dont remember where..

**Reply by ederbiason** [reply]
Unfortunately not, I used a little scrollbar at the bottom of the section like @sorasful...

**Comment by Brianrahmarela** [comment]
**in jsx:**

```
<div className="rounded-lg scrollxshow">
  <div>
	//your item
  </div>			
</div>
```

**in global.css:**

div.scrollxshow {
  overflow: auto;
}

div.scrollxshow div {
  display: inline-block;
}
